python_modules/dagster/dagster/_core/definitions/parent_asset_graph_differ.py

```diff
@@ -0,0 +1,144 @@
+from enum import Enum
+from typing import TYPE_CHECKING, Callable, Optional, Sequence, Union
+
+import dagster._check as check
+from dagster._core.definitions.external_asset_graph import ExternalAssetGraph
+from dagster._core.errors import DagsterInvariantViolationError
+from dagster._core.host_representation import ExternalRepository
+from dagster._core.workspace.context import BaseWorkspaceRequestContext
+
+if TYPE_CHECKING:
+    from dagster._core.definitions.events import (
+        AssetKey,
+    )
+
+
+class ChangeReason(Enum):
+    NEW = "NEW"
+    CODE_VERSION = "CODE_VERSION"
+    INPUTS = "INPUTS"
```

Comment:
yeah i wasn't quite sure what would count as a change in output since each asset key should only have one output. we could inspect the type of the output maybe? it seemed worthy of more consideration so i didn't include it in this pass